### PR TITLE
EXPOSERS: Delete Team By Id

### LINKED ISSUE #126: EXPOSERS: Delete `Team`

### PR DESCRIPTION
**Pull request by iamDiyorjon**
closes #126

### CODE REVIEW

File: Tarteeb.Api/Controllers/TeamsController.cs [hunk 1]
```diff
@@ -78,5 +78,43 @@ public async ValueTask<ActionResult<Team>> GetTeamByIdAsync(Guid Id)
...
+                return Ok(deletedTeam);
+            }
+            catch (TeamValidationException teamValidationException)
+                when(teamValidationException.InnerException is NotFoundTeamException)
+            {
+                return NotFound(teamValidationException.InnerException);
+            }
+            catch(TeamValidationException teamValidationException)
+            {
+                return BadRequest(teamValidationException);
```
Comment (jamshidbek001) [inline Tarteeb.Api/Controllers/TeamsController.cs:99]: innerException?

File: Tarteeb.Api/Controllers/TeamsController.cs [hunk 2]
```diff
@@ -78,5 +78,43 @@ public async ValueTask<ActionResult<Team>> GetTeamByIdAsync(Guid Id)
...
+            }
+            catch(TeamDependencyValidationException teamDependencyValidationException)
+                when(teamDependencyValidationException.InnerException is LockedTeamException)
+            {
+                return Locked(teamDependencyValidationException);
```
Comment (jamshidbek001) [inline Tarteeb.Api/Controllers/TeamsController.cs:104]: innerExveption?

File: Tarteeb.Api/Controllers/TeamsController.cs [hunk 3]
```diff
@@ -78,5 +78,43 @@ public async ValueTask<ActionResult<Team>> GetTeamByIdAsync(Guid Id)
...
+            {
+                return BadRequest(teamDependencyValidationException);
+            }
+            catch(TeamDependencyException teamDependencyException)
+            {
+                return InternalServerError(teamDependencyException);
+            }
+            catch(TeamServiceException teamServiceException)
+            {
+                return InternalServerError(teamServiceException.InnerException);
```
Comment (jamshidbek001) [inline Tarteeb.Api/Controllers/TeamsController.cs:116]: needless InnerException

**Review by jamshidbek001 — CHANGES_REQUESTED** [review]
Check the parametres

File: Tarteeb.Api/Controllers/TeamsController.cs [hunk 4]
```diff
@@ -78,5 +78,43 @@ public async ValueTask<ActionResult<Team>> GetTeamByIdAsync(Guid Id)
...
             }
         }
+
+        [HttpDelete("{teamId}")]
+        public async ValueTask<ActionResult<Team>>DeleteTeamByIdAsync(Guid teamId)
+        {
+            try
+            {
+                Team deletedTeam = 
+                    await this.teamService.RetrieveTeamByIdAsync(teamId);
```
Comment (jamshidbek001) [inline Tarteeb.Api/Controllers/TeamsController.cs:88]: RemoveTeamByIdAsync bo'lishi kerak

**Review by jamshidbek001 — CHANGES_REQUESTED** [review]
Use right Method

**Review by StudentofTATU — CHANGES_REQUESTED** [review]
Ctrl + k +e

File: Tarteeb.Api/Controllers/TeamsController.cs [hunk 5]
```diff
@@ -78,5 +78,43 @@ public async ValueTask<ActionResult<Team>> GetTeamByIdAsync(Guid Id)
...
+                return BadRequest(teamDependencyValidationException);
+            }
+            catch(TeamDependencyException teamDependencyException)
+            {
+                return InternalServerError(teamDependencyException);
+            }
+            catch(TeamServiceException teamServiceException)
+            {
+                return InternalServerError(teamServiceException);
+            }
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Controllers/TeamsController.cs:117]: Inner Exceptions

**Review by ElbekDeveloper — APPROVED** [review]

**Review by ElbekDeveloper — APPROVED** [review]

**Pull request merged**